### PR TITLE
Deployment script for contracts-periphery and faucet

### PR DESCRIPTION
**Pull request by tremarkley**
Part of https://github.com/ethereum-optimism/ecopod/issues/533
Part of https://github.com/ethereum-optimism/ecopod/issues/534 

This PR implements the following:
- A deployment system for deploying the contracts inside of the contracts-periphery directory. 
- A deployment script for deploying an upgradeable faucet contract

For UX reasons we would like the `FaucetProxy` contract to be deployed at the same address across all OP Chains. In order to do that the `ProxyAdmin`, `FaucetProxy` must provide a `salt` on contract creation.

### CODE REVIEW

**Comment by tremarkley** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #7203** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/7203" target="_blank">Graphite</a>  👈
    * **PR #7342** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/7342" target="_blank">Graphite</a> 
      * **PR #7317** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/7317" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/7203?utm_source=stack-comment).

File: packages/contracts-bedrock/src/periphery/scripts/DeployPeriphey.s.sol [hunk 1]
```diff
@@ -0,0 +1,10 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
```
Comment (tynes) [inline packages/contracts-bedrock/src/periphery/scripts/DeployPeriphey.s.sol]: This script should go in `contracts-bedrock/scripts`

File: packages/contracts-bedrock/scripts/DeployPeripheryConfig.s.sol [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import { Script } from "forge-std/Script.sol";
+import { console2 as console } from "forge-std/console2.sol";
+import { stdJson } from "forge-std/StdJson.sol";
+
+/// @title DeployPeripheryConfig
+/// @notice Represents the configuration required to deploy the periphery contracts. It is expected
```
Comment (tynes) [inline packages/contracts-bedrock/scripts/DeployPeripheryConfig.s.sol]: I think this reads a bit strange and would read better as `PeripheryDeployConfig.s.sol` but i do see the benefit of having the files named alphabetically similarly

**Review by tynes — APPROVED** [review]

File: packages/contracts-bedrock/scripts/DeployPeriphery.s.sol [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+        save("FaucetProxy", address(proxy));
+        console.log("FaucetProxy deployed at %s", address(proxy));
+
+        addr_ = address(proxy);
+    }
+
+    /// @notice Deploy the faucet contract.
+    function deployFaucet() public broadcast returns (address) {
+        Faucet faucet = new Faucet{ salt: FAUCET_IMPL_SALT }(cfg.faucetAdmin());
+
```
Comment (tynes) [inline packages/contracts-bedrock/scripts/DeployPeriphery.s.sol:96]: Is there a getter for the faucet admin? Would be nice to have a require for it here

File: packages/contracts-bedrock/scripts/DeployPeriphery.s.sol [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        return address(faucet);
+    }
+
+    /// @notice Initialize the Faucet
+    function initializeFaucet() public broadcast {
+        ProxyAdmin proxyAdmin = ProxyAdmin(mustGetAddress("ProxyAdmin"));
+        address faucetProxy = mustGetAddress("FaucetProxy");
+        address faucet = mustGetAddress("Faucet");
+
+        proxyAdmin.upgrade({ _proxy: payable(faucetProxy), _implementation: faucet });
```
Comment (tynes) [inline packages/contracts-bedrock/scripts/DeployPeriphery.s.sol:109]: Would be nice to have a require on the faucet admin here

File: packages/contracts-bedrock/scripts/DeployPeripheryConfig.s.sol (hunk 2 above)
Comment (tremarkley) [reply packages/contracts-bedrock/scripts/DeployPeripheryConfig.s.sol]: sounds good, I updated the name to `PeripheryDeployConfig.s.sol`

File: packages/contracts-bedrock/scripts/DeployPeriphery.s.sol (hunk 3 above)
Comment (tremarkley) [reply packages/contracts-bedrock/scripts/DeployPeriphery.s.sol:96]: done

File: packages/contracts-bedrock/scripts/DeployPeriphery.s.sol (hunk 4 above)
Comment (tremarkley) [reply packages/contracts-bedrock/scripts/DeployPeriphery.s.sol:109]: done

**Pull request merged**